File: backend/app/schemas/lead.py
from pydantic import BaseModel

from app.schemas.user import UserRead


class LeadEmailRead(BaseModel):
    id: str
    email: str
    source: str
    confidence: float
    mx_valid: bool | None = None
    role_based: bool = False
    disposable: bool = False
    smtp_verified: bool | None = None
    validated_at: int | None = None

    model_config = {"from_attributes": True}


class LeadContactRead(BaseModel):
    id: str
    name: str | None
    title: str | None
    phone: str | None
    email: str | None
    source: str

    model_config = {"from_attributes": True}


class TagRead(BaseModel):
    id: str
    name: str
    color: str | None = None

    model_config = {"from_attributes": True}


class LeadRead(BaseModel):
    id: str
    place_id: str | None
    name: str
    address: str | None
    city: str | None
    state: str | None
    phone: str | None
    phone_normalized: str | None = None
    phone_type: str | None = None
    website: str | None
    place_types: list[str]
    emails: list[LeadEmailRead] = []
    contacts: list[LeadContactRead] = []
    tags: list[TagRead] = []
    employee_count_min: int | None
    employee_count_max: int | None
    revenue_range: str | None
    location_count: int | None
    status: str
    notes: str | None
    source: str
    scraped_at: int | None
    assigned_to_user_id: str | None = None
    assignee: UserRead | None = None
    last_touched_at: int | None = None
    last_touched_by_user_id: str | None = None
    last_touched_by: UserRead | None = None
    score: int | None = None
    matched_segment_ids: list[str] = []
    summary: str | None = None
    summary_generated_at: int | None = None
    fit_reasons: list[dict] = []
    fit_reasons_generated_at: int | None = None
    created_at: int
    updated_at: int

    model_config = {"from_attributes": True}


class LeadUpdate(BaseModel):
    status: str | None = None
    notes: str | None = None
    employee_count_min: int | None = None
    employee_count_max: int | None = None
    revenue_range: str | None = None
    name: str | None = None
    website: str | None = None
    phone: str | None = None
    address: str | None = None
    city: str | None = None
    state: str | None = None
    place_types: list[str] | None = None


class LeadAssign(BaseModel):
    user_id: str | None = None  # null = unassign

File: frontend/src/types/lead.ts
export type LeadStatus = 'new' | 'contacted' | 'qualified' | 'rejected'

import type { User } from './user'
import type { Tag } from './tag'

export interface LeadEmail {
  id: string
  email: string
  source: string
  confidence: number
  mx_valid?: boolean | null
  role_based?: boolean
  disposable?: boolean
  validated_at?: number | null
}

export interface LeadContact {
  id: string
  name: string | null
  title: string | null
  phone: string | null
  email: string | null
  source: string
}

export interface Lead {
  id: string
  place_id: string | null
  name: string
  address: string | null
  city: string | null
  state: string | null
  phone: string | null
  phone_normalized?: string | null
  phone_type?: string | null
  website: string | null
  place_types: string[]
  emails: LeadEmail[]
  contacts: LeadContact[]
  tags: Tag[]
  employee_count_min: number | null
  employee_count_max: number | null
  revenue_range: string | null
  location_count: number | null
  status: LeadStatus
  notes: string | null
  source: string
  scraped_at: number | null
  assigned_to_user_id?: string | null
  assignee?: User | null
  last_touched_at?: number | null
  last_touched_by_user_id?: string | null
  last_touched_by?: User | null
  score?: number | null
  fit_score?: number | null
  intent_score?: number | null
  priority_score?: number | null
  score_breakdown?: Record<string, unknown>
  matched_segment_ids?: string[]
  summary?: string | null
  summary_generated_at?: number | null
  fit_reasons?: FitReason[]
  fit_reasons_generated_at?: number | null
  signals?: LeadSignal[]
  created_at: number
  updated_at: number
}

export interface LeadSignal {
  id: string
  type: string
  strength: number
  source: string
  payload: Record<string, unknown>
  detected_at: number
}

export interface FitReason {
  segment_id: string
  segment_name: string
  rationale: string
}

export interface LeadsPage {
  total: number
  page: number
  page_size: number
  items: Lead[]
}

export interface LeadUpdate {
  status?: LeadStatus
  notes?: string
  employee_count_min?: number
  employee_count_max?: number
  revenue_range?: string
  name?: string
  website?: string | null
  phone?: string | null
  address?: string | null
  city?: string | null
  state?: string | null
  place_types?: string[]
}

export interface LeadActivity {
  id: string
  lead_id: string
  user_id: string | null
  user?: User | null
  action: string
  payload: Record<string, unknown>
  created_at: number
}
